### ISSUE TITLE
Support for team-based permissions that span multiple organizations (either centralize or copy or allow invites to multiple people at once)

### ISSUE THREAD

**Issue by jmcmurry** [isaacs/github#953]
## TL;DR 
GitHub `teams`, much like GitHub `users`, often _functionally_ span multiple GitHub `organizations`, however, unlike `users`, `teams` are currently technically bound to `organizations`. Better support for cross-`org` `teams` would be appreciated. 

## Background
My team of ~40 spans 8 universities;  the GitHub `org` construct is lacking. The work we do is all interdependent and matricised -- broken down into 50+ repos across 6 different GitHub `orgs`. Managing access in this context is arduous because we have to have mirrored `teams` across the 6 different GitHub `orgs`. These interdependent efforts span many years and it makes no sense to shift them to new `orgs` where they could be centrally managed, only to move them to yet another `org` a few years later under different funding etc. The rest of the github `org`/`repo` structure works pretty well for us, especially now that it is possible to co-opt the [new repo-topics feature](https://github.com/blog/2309-introducing-topics). Thus the pain of managing team-wide discoverability of repos is now limited to private repos. At least in academia, this dynamic and complex scenario is not as rare as you might think. 

## Best solution: Teams independent of orgs, or shared between orgs
Support for a single `org`-INDEPENDENT team, managed with one primary `org` as owner, and other `orgs` as managers of the `team`. Thus when a member joins or leaves the overall effort [straining here to use the word 'effort' rather than terms like 'project' as they have specific meaning in github], their access to the various repos in the various `orgs` could be managed once rather than six times. This is technically more difficult to achieve.

## Second-best solution: Copy permissions from one org to another
Support for copy an entire `team` permissions from one `org` to another. This would require no changes to the existing structure of permissions, but would still reduce tedium and error on the part of those managing. 

## Third-best solution: Invite multiple users at once
Support for inviting multiple github users to a team at once. While this wouldn't help with the case where permissions must be removed from someone who has left, it would be better than doing forty invites one-by-one for each new org as we grow. This feature might be useful to others with different use cases.

## Fourth-best solution: Script updates using GitHub API
I'm aware we can use the [github API to do the team mirroring](https://developer.github.com/v3/orgs/teams/#remove-team-member). The reason this isn't ideal is that: 
A) Matching on string of team names is dangerous (it would need to be a super unique team name that is impossible to be chosen by accident).
B) We would need to designate a superuser as `owner` on all the `orgs` -- this is overkill for the functionality we need, and a bit dangerous.

Nevertheless, the pseudocode might look something like this:
````
String `master_org`  // URL for the org designated the 'master' org
Set `org_list`  // A list of all other organization URLs 
String `master_team_name` // Name of master team

for each org in org_list
 for each team_name in org
    if team_name equals master_team_name
      for each user in org.team
          if user exists in master.team
                   copy user permissions to user in org.team
          else remove user from org.team
````
If anyone has a script that already does something like this, let me know how it is going? Possibly share (anonymized) version of it?

**Comment by immo-huneke-zuhlke** [comment]
I was looking for exactly this - the ability to define members of a team spanning more than one organization. I am surprised that this concept is not already supported out of the box. It would clearly help where people from multiple organizations are working together on some programme of work, but also where an individual belongs to more than one organization (consecutively or simultaneously), which happens all the time.

**Comment by immo-huneke-zuhlke** [comment]
If it's too difficult to extend the concept of teams (which originally were simply a way of grouping members of an organization into smaller subsets), how about introducing a new grouping concept such as "program"? A program in common parlance is a co-ordinated set of projects and, by extension, the set of people working on those projects. This idea could be used as a cross-organizational way to manage access permissions, including access to the relevant project boards and repositories of the member organizations.
